grails-app/domain/org/pih/warehouse/order/Order.groovy

```diff
@@ -176,10 +178,13 @@ class Order implements Serializable {
         return (status == OrderStatus.CANCELED)
     }
 
-    def listShipments() {
+    def getShipments() {
         return orderItems.collect { it.listShipments() }.flatten().unique() { it?.id }
     }
 
+    List getShipmentsByStatus(ShipmentStatusCode statusCode) {
+        return getShipments().findAll { it -> it.currentStatus == statusCode }
```

Comment:
You can just use `shipments.findAll { ... }` here.

Reply:
Fixed